### PR TITLE
modify template to use nf-validation plugin

### LINKED ISSUE #1543: Port schema validation to core Nextflow (plugin)
### Description of feature

At the moment, validation of `nextflow_schema.json` files relies on code held in the nf-core template, including dependencies in the `lib` folder.

Now that this functionality has matured it is relatively stable and is now used in many pipelines beyond the nf-core ecosystem. It would be great to move this to core Nextflow functionality for more widespread usage and maintenance.

This was already proposed last year in this issue: https://github.com/nextflow-io/nextflow/issues/2434 and Paolo responded by creating this Nextflow plugin: https://github.com/nextflow-io/nf-validation

The code there is copied from the nf-core template at the time. It needs updating to be in line with the current template, and polishing (checking the test functionality, documentation). We also need to figure out how to actually use it within nf-core pipelines and finally strip out all of this code from the template.

### PR DESCRIPTION
**Pull request by mirpedrol**
Closes #1543 

The validation of nextflow_schema.json files from nf-core pipelines is now part of a core Nextflow plugin: https://github.com/nextflow-io/nf-validation/pull/5

Test in [nf-core/testpipeline](https://github.com/nf-core/testpipeline/tree/nf-validation-plugin)

Functions can be imported to a pipeline with `include { paramsHelp, paramsSummaryMap, paramsSummaryLog } from 'plugin/nf-validation'`

Schema validation code from the pipeline template `NfcoreSchema.groovy` and `MainWorkflow.groovy` is removed.

Validate a sample sheet with a JSON schema and create a channel from it. Replaces `check_samplesheet` subworkflow and code.

## PR checklist

- [ ] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated

### CODE REVIEW

File: nf_core/pipeline-template/workflows/pipeline.nf [hunk 1]
```diff
@@ -4,11 +4,31 @@
...
+include { paramsSummaryMap   } from 'plugin/nf-validation'
+
+def logo = NfcoreTemplate.logo(workflow, params.monochrome_logs)
+def citation = '\n' + WorkflowMain.citation(workflow) + '\n'
+def summary_params = paramsSummaryMap(workflow)
+
+// Print help message if needed
+if (params.help) {
+    def String command = "nextflow run ${workflow.manifest.name} --input samplesheet.csv --genome GRCh37 -profile docker"
+    log.info logo + paramsHelp(command) + citation += NfcoreTemplate.dashedLine(params.monochrome_logs)
```
Comment (mirpedrol) [inline nf_core/pipeline-template/workflows/pipeline.nf:19]:
```suggestion
    log.info logo + paramsHelp(command) + citation + NfcoreTemplate.dashedLine(params.monochrome_logs)
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,8 @@
 - Move registry definitions out of profile scope ([#2286])(https://github.com/nf-core/tools/pull/2286)
 - Remove `aws_tower` profile ([#2287])(https://github.com/nf-core/tools/pull/2287)
 - Fixed the Slack report to include the pipeline name ([#2291](https://github.com/nf-core/tools/pull/2291))
+- Remove shcema validation from `lib` folder and use Nextflow nf-validation plugin instead ([#1771](https://github.com/nf-core/tools/pull/1771/))
```
Comment (mashehu) [inline CHANGELOG.md:14]:
```suggestion
- Remove schema validation from `lib` folder and use Nextflow nf-validation plugin instead ([#1771](https://github.com/nf-core/tools/pull/1771/))
```

File: nf_core/pipeline-template/nextflow.config [hunk 3]
```diff
@@ -58,6 +56,13 @@ params {
     max_cpus                   = 16
     max_time                   = '240.h'
 
+    // Schema validation default options
+    validationFailUnrecognisedParams = false
+    validationLenientMode            = false
+    validate_params                  = true
```
Comment (mashehu) [inline nf_core/pipeline-template/nextflow.config:62]: Should we also make them camel-case?

**Review by mashehu — APPROVED** [review]

File: nf_core/pipeline-template/nextflow.config (hunk 3 above)
Comment (mirpedrol) [reply nf_core/pipeline-template/nextflow.config:62]: I'm doubting with `validate_params`. For `schema_ignore_params ` you are right, it should be camelCase as it's part of the nf-validation plugin. But `validate_params` is only used in nf-core pipelines, so I think it should be snake_case?

Comment (maxulysse) [reply nf_core/pipeline-template/nextflow.config:62]: I'd go for all `snake_case`

Comment (maxulysse) [reply nf_core/pipeline-template/nextflow.config:62]: if it's params, then `snake_case`

Comment (mirpedrol) [reply nf_core/pipeline-template/nextflow.config:62]: Unfortunately the nf-validation plugin is using camelCase as nextflow uses this notation, and it's ment to be for all NF pipelines and not only nf-core.

Comment (maxulysse) [reply nf_core/pipeline-template/nextflow.config:62]: Good point, then ok for camelCase

**Comment by mirpedrol** [comment]
Converting it to a draft again as, after some Slack discussions, we've decided to not include `fromSamplesheet()` yet.

**Comment by mirpedrol** [comment]
I messed up this PR too much, so I've decided to open a new one https://github.com/nf-core/tools/pull/2300

**Pull request closed without merging**